### PR TITLE
Move from npm to Yarn for markdown tests

### PR DESCRIPTION
**Pull request by RDIL**
# PR Summary
Yarn is better because...
* it validates package checksums
* it has emojis 🤟
* it is faster and employs better caching

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: test/common/markdown/markdown.tests.ps1 [hunk 1]
```diff
@@ -17,7 +17,7 @@ Describe 'Common Tests - Validate Markdown Files' -Tag 'CI' {
         {
             $NpmInstalled = "Installed"
             Write-Verbose -Message "NPM is checking Gulp is installed. This may take a few moments." -Verbose
-            start-nativeExecution { npm install --silent }
+            start-nativeExecution { yarn }
             start-nativeExecution { npm install 'gulp@4.0.0' --silent }
```
Comment (TravisEz13) [inline test/common/markdown/markdown.tests.ps1]: If we are going to use yarn, can we use the yarn commands?

File: .vsts-ci/misc-analysis.yml [hunk 2]
```diff
@@ -36,6 +36,11 @@ jobs:
     displayName: Install Pester
     condition: succeededOrFailed()
 
+  - bash: |
+      curl -o- --progress-bar -L https://yarnpkg.com/install.sh | bash
+    displayName: Bootstrap Yarn
+    condition: succeededOrFailed()
+
   - bash: |
       sudo npm install -g markdown-spellcheck@0.11.0
```
Comment (TravisEz13) [inline .vsts-ci/misc-analysis.yml]: If we are going to use yarn, can we use the yarn commands?

File: test/common/markdown/markdown.tests.ps1 (hunk 1 above)
Comment (TravisEz13) [reply test/common/markdown/markdown.tests.ps1]: line 25 too

Comment (RDIL) [reply test/common/markdown/markdown.tests.ps1]: Good point.  Not sure why this is here anyway as it should be installed by package.json.

File: .vsts-ci/misc-analysis.yml (hunk 2 above)
Comment (RDIL) [reply .vsts-ci/misc-analysis.yml]: Can’t install globally with yarn.  This needs to stay.

File: test/common/markdown/markdown.tests.ps1 (hunk 1 above)
Comment (TravisEz13) [reply test/common/markdown/markdown.tests.ps1]: It should... I added it because it wasn't always installed.

Comment (RDIL) [reply test/common/markdown/markdown.tests.ps1]: Replaced it with yarn add.

File: .vsts-ci/misc-analysis.yml (hunk 2 above)
Comment (TravisEz13) [reply .vsts-ci/misc-analysis.yml]: Can you add a comment saying this here and on line 25 in the test?

Comment (RDIL) [reply .vsts-ci/misc-analysis.yml]: Okay

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 16 hours

**Comment by PoshChan** [comment]
@TravisEz13, will remind you in 16 hours

**Comment by RDIL** [comment]
@TravisEz13 this is your 16 hour reminder

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 16 hours ago

**Comment by RDIL** [comment]
@TravisEz13 ping

**Pull request merged**